Plugin crashes when trying to import cloudflare_custom_ssl

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.0.7
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v3.1.0
+ provider registry.terraform.io/hashicorp/aws v3.61.0
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/vault v2.24.0

### Affected resource(s)

cloudflare_custom_ssl

### Terraform configuration files

```hcl
resource "cloudflare_custom_ssl" "multisan_certificate" {
  count = var.enable_custom_ssl ? 1 : 0
  zone_id  = var.zone_id
  custom_ssl_options  {
    certificate = data.vault_generic_secret.cert.data["tls.crt"]
    private_key = data.vault_generic_secret.cert.data["tls.key"]
    type = "sni_custom"
  }
}
```


### Debug output

2021-10-05T12:29:00.730+0200 [INFO]  provider.terraform-provider-cloudflare_v3.1.0: 2021/10/05 12:29:00 [DEBUG] Cloudflare API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Cf-Cache-Status: DYNAMIC
Cf-Ray: 6995f761f8050111-AMS
Content-Type: application/json
Date: Tue, 05 Oct 2021 10:29:00 GMT
Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Expires: Sun, 25 Jan 1981 05:00:00 GMT
Pragma: no-cache
Server: cloudflare
Set-Cookie: __cflb=0H28vgHxwvgAQtjUGU56Rb8iNWZVUvXhchhLhV7wucy; SameSite=Lax; path=/; expires=Tue, 05-Oct-21 12:59:01 GMT; HttpOnly
Set-Cookie: __cfruid=6455d15568002ed2b3fabe9c5ebfdbf2d2f8a6b4-1633429740; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
Strict-Transport-Security: max-age=31536000
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN

{
 "result": {
  "id": "",
  "type": "sni_custom",
  "hosts": [
   "domain.nl",
   "domain.pl",
   "domain.de",
   "domain.at",
   "domain.it",
   "domain.no",
   "domain.ie",
   "domain.fr",
   "domain.es",
   "domain.dk",
   "domain.ch",
   "domain.ch",
   "domain.com",
   "*.domain.nl",
   "*.domain.com",
   "*.domain.pl",
   "*.domain.de",
   "*.domain.at",
   "*.domain.it",
   "*.domain.no",
   "*.domain.ie",
   "*.domain.fr",
   "*.domain.es",
   "*.domain.dk",
   "*.domain.ch",
   "*.domain.ch",
   "*.domain.com",
   "domain.com"
  ],
  "issuer": "SectigoLimited",
  "signature": "SHA256WithRSA",
  "bundle_method": "ubiquitous",
  "zone_id": "[REDACTED]",
  "uploaded_on": "[REDACTED]",
  "modified_on": "[REDACTED]",
  "expires_on": "[REDACTED]",
  "priority": 0,
  "status": "active"
 },
 "success": true,
 "errors": [],
 "messages": []
}
-----------------------------------------------------: timestamp=2021-10-05T12:29:00.729+0200
2021-10-05T12:29:00.730+0200 [INFO]  provider.terraform-provider-cloudflare_v3.1.0: 2021/10/05 12:29:00 [DEBUG] Custom SSL options found in config: []interface {}{}: timestamp=2021-10-05T12:29:00.730+0200
2021-10-05T12:29:00.730+0200 [INFO]  provider.terraform-provider-cloudflare_v3.1.0: 2021/10/05 12:29:00 [DEBUG] Custom SSL JSON: {}: timestamp=2021-10-05T12:29:00.730+0200
2021-10-05T12:29:00.730+0200 [INFO]  provider.terraform-provider-cloudflare_v3.1.0: 2021/10/05 12:29:00 [DEBUG] Custom SSL options crdomaining: cloudflare.ZoneCustomSSLOptions{Certificate:"", PrivateKey:"", BundleMethod:"", GeoRestrictions:(*cloudflare.ZoneCustomSSLGeoRestrictions)(nil), Type:""}: timestamp=2021-10-05T12:29:00.730+0200
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: PANIC: runtime error: invalid memory address or nil pointer dereference
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xd45f8e]
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: 
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: goroutine 45 [running]:
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenCustomSSLOptions(...)
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:360
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareCustomSslRead(0xc0002da480, {0xfdb3c0, 0xc0007c23c0})
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:233 +0x52e
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareCustomSslImport(0xc0002da480, {0xfdb3c0, 0xc0007c23c0})
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:278 +0x2f9
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc000378640, {0x1206248, 0xc0002ae100}, 0xc00066bae8, {0xc0000421e0, 0x45})
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/provider.go:375 +0x142
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc000342180, {0x1206248, 0xc0002ae100}, 0xc000300140)
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:1021 +0xdf
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ImportResourceState(0xc000301980, {0x12062f0, 0xc000722300}, 0xc0003781e0)
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/server/server.go:349 +0xb5
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler({0xf9dbc0, 0xc000301980}, {0x12062f0, 0xc000722300}, 0xc00072c180, 0x0)
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:398 +0x170
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00033b180, {0x1217c28, 0xc00027a000}, 0xc0002fe000, 0xc0001d5d40, 0x18b0e78, 0x0)
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: google.golang.org/grpc.(*Server).handleStream(0xc00033b180, {0x1217c28, 0xc00027a000}, 0xc0002fe000, 0x0)
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: google.golang.org/grpc.(*Server).serveStreams.func1.2()
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     google.golang.org/grpc@v1.40.0/server.go:941 +0x98
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0: crdomained by google.golang.org/grpc.(*Server).serveStreams.func1
2021-10-05T12:29:00.732+0200 [DEBUG] provider.terraform-provider-cloudflare_v3.1.0:     google.golang.org/grpc@v1.40.0/server.go:939 +0x294
2021-10-05T12:29:00.734+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.1.0/linux_amd64/terraform-provider-cloudflare_v3.1.0 pid=1062826 error="exit status 2"
2021-10-05T12:29:00.734+0200 [ERROR] plugin.(*GRPCProvider).ImportResourceState: error="rpc error: code = Unavailable desc = transport is closing"
2021-10-05T12:29:00.734+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ImportResourceState call. The plugin logs may
│ contain more details.
╵

2021-10-05T12:29:00.735+0200 [DEBUG] [aws-sdk-go] DEBUG: Request dynamodb/GetItem Details:
---[ REQUEST POST-SIGN ]-----------------------------
POST / HTTP/1.1
Host: dynamodb.eu-west-1.amazonaws.com
User-Agent: aws-sdk-go/1.40.25 (go1.16.4; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/1.0.7
Content-Length: 195
Accept-Encoding: identity
Authorization: [REDACTED]
Content-Type: application/x-amz-json-1.0
X-Amz-Date: 20211005T102900Z
X-Amz-Target: DynamoDB_20120810.GetItem

{"ConsistentRead":true,"Key":{"LockID":{"S":"terraform-state-store/cdn/cloudflare/prod/domain.at/terraform.tfstate"}},"ProjectionExpression":"LockID, Info","TableName":"terraform-state-lock"}
-----------------------------------------------------
2021-10-05T12:29:01.054+0200 [DEBUG] [aws-sdk-go] DEBUG: Response dynamodb/GetItem Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Content-Length: 415
Content-Type: application/x-amz-json-1.0
Date: Tue, 05 Oct 2021 10:29:00 GMT
Server: Server
X-Amz-Crc32: 1687081459
X-Amzn-Requestid: 68I0H90OFU8H02KEOTOGSCAS7RVV4KQNSO5AEMVJF66Q9ASUAAJG


-----------------------------------------------------
2021-10-05T12:29:01.054+0200 [DEBUG] [aws-sdk-go] {"Item":{"LockID":{"S":"terraform-state-store/cdn/cloudflare/prod/domain.at/terraform.tfstate"},"Info":{"S":"{\"ID\":\"1c571667-8a94-340d-18bb-6fb0bbd48629\",\"Operation\":\"OperationTypeInvalid\",\"Info\":\"\",\"Who\":\"SHA256 Credential=AKIA3QDFVGEZMB45CXF3/20211005/eu-west-1/dynamodb/aws4_request, SignedHeaders=accept-encoding;content-length;content-type;host;x-amz-date;x-amz-target, Signature=faa0e590caad161e3faea12c9d142e7018546399e2141e0b04422a53799f7dd2\",\"Version\":\"1.0.7\",\"Crdomained\":\"2021-10-05T10:28:35.441146286Z\",\"Path\":\"terraform-state-store/cdn/cloudflare/prod/domain.at/terraform.tfstate\"}"}}}
2021-10-05T12:29:01.055+0200 [DEBUG] [aws-sdk-go] DEBUG: Request dynamodb/DeleteItem Details:
---[ REQUEST POST-SIGN ]-----------------------------
POST / HTTP/1.1
Host: dynamodb.eu-west-1.amazonaws.com
User-Agent: aws-sdk-go/1.40.25 (go1.16.4; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/1.0.7
Content-Length: 135
Accept-Encoding: identity
Authorization: [REDACTED]
Content-Type: application/x-amz-json-1.0
X-Amz-Date: 20211005T102901Z
X-Amz-Target: DynamoDB_20120810.DeleteItem

{"Key":{"LockID":{"S":"terraform-state-store/cdn/cloudflare/prod/domain.at/terraform.tfstate"}},"TableName":"terraform-state-lock"}
-----------------------------------------------------
2021-10-05T12:29:01.336+0200 [DEBUG] [aws-sdk-go] DEBUG: Response dynamodb/DeleteItem Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Content-Length: 2
Content-Type: application/x-amz-json-1.0
Date: Tue, 05 Oct 2021 10:29:01 GMT
Server: Server
X-Amz-Crc32: 2745614147
X-Amzn-Requestid: ENGHUBO01UPROCTARBJS4QAPUVVV4KQNSO5AEMVJF66Q9ASUAAJG


-----------------------------------------------------
2021-10-05T12:29:01.336+0200 [DEBUG] [aws-sdk-go] {}
2021-10-05T12:29:01.336+0200 [DEBUG] provider: plugin exited
ERRO[0029] 1 error occurred:
        * exit status 1
         


### Panic output

Stack trace from the terraform-provider-cloudflare_v3.1.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xd45f8e]

goroutine 11 [running]:
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenCustomSSLOptions(...)
        github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:360
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareCustomSslRead(0xc000188680, {0xfdb3c0, 0xc0005c43c0})
        github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:233 +0x52e
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareCustomSslImport(0xc000188680, {0xfdb3c0, 0xc0005c43c0})
        github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:278 +0x2f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc0006146e0, {0x1206248, 0xc0003455c0}, 0xc00011bae8, {0xc00059e0f0, 0x45})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/provider.go:375 +0x142
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc000650108, {0x1206248, 0xc0003455c0}, 0xc0008f23a0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:1021 +0xdf
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ImportResourceState(0xc0005818a0, {0x12062f0, 0xc0008f84b0}, 0xc0008fe370)
        github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/server/server.go:349 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler({0xf9dbc0, 0xc0005818a0}, {0x12062f0, 0xc0008f84b0}, 0xc0008bfb00, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:398 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000426a80, {0x1217c28, 0xc00027c600}, 0xc00064b320, 0xc00024d7d0, 0x18b0e78, 0x0)
        google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000426a80, {0x1217c28, 0xc00027c600}, 0xc00064b320, 0x0)
        google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.40.0/server.go:939 +0x294

Error: The terraform-provider-cloudflare_v3.1.0 plugin crashed!


### Expected output

Resourse imported successfully

### Actual output

Plugin crashed

### Steps to reproduce

      terragrunt import 'cloudflare_custom_ssl.multisan_certificate[0]' '[REDACTED]'

### Additional factoids

As I was trying to import my certificates to terraform, my plugin kept crashing. I'm not sure if it has anything to do with the fact that terraform fetches the certificate info from Vault, but it is unable to perform the import, as you can see from the debug output.

### References

_No response_

Hello there,

We are, unfortunately, presented with the exact same issue as the one mentioned above in its most simplistic form so I am providing some more info here for your convenience.

We have a custom SSL cert resource defined as follows

```
resource "cloudflare_custom_ssl" "asdf_ssl_cert" {
  zone_id = var.asdf_zone_id
  custom_ssl_options {
    certificate = var.asdf_ssl_cert
    private_key = var.asdf_ssl_key
    bundle_method = "ubiquitous"
    type = "sni_custom"
  }
}
```

When trying to import an existing SSL custom cert into terraform using the following command : 
`terraform import cloudflare_custom_ssl.asdf_ssl_cert [zone_id]/[ssl cert resource id]`

We get the following error : 

```
Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ImportResourceState call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-cloudflare_v3.8.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x16f416e]

goroutine 24 [running]:
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenCustomSSLOptions(...)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:269
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareCustomSslRead(0xc00030ec80, {0x1921a60, 0xc0000b43c0})
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:142 +0x52e
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareCustomSslImport(0xc00030ec80, {0x1921a60, 0xc0000b43c0})
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_custom_ssl.go:187 +0x2f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc00007f7a0, {0x1a55e78, 0xc0000c6340}, 0xc000595a28, {0xc00031a050, 0x45})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/provider.go:384 +0x142
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc00000d398, {0x1a55e78, 0xc0000c6340}, 0xc00030a500)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:1043 +0xdf
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ImportResourceState(0xc000181080, {0x1a55f20, 0xc0001b41e0}, 0xc0001b62d0)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:628 +0x262
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler({0x18f66e0, 0xc000181080}, {0x1a55f20, 0xc0001b41e0}, 0xc000012180, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:398 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002aac40, {0x1a646e8, 0xc000510000}, 0xc0001305a0, 0xc0004115c0, 0x1f9c418, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0002aac40, {0x1a646e8, 0xc000510000}, 0xc0001305a0, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.40.0/server.go:939 +0x294

Error: The terraform-provider-cloudflare_v3.8.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I am also having the same issue and I am not able to import custom certificates. Error is at follows

```
Stack trace from the terraform-provider-cloudflare_v3.25.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xbd3bbf]

goroutine 81 [running]:
github.com/cloudflare/terraform-provider-cloudflare/internal/provider.flattenCustomSSLOptions(...)
	github.com/cloudflare/terraform-provider-cloudflare/internal/provider/resource_cloudflare_custom_ssl.go:278
github.com/cloudflare/terraform-provider-cloudflare/internal/provider.resourceCloudflareCustomSslRead({0x1013fe0, 0xc000728a80}, 0xc0000fa300, {0xe93f00?, 0xc000328480})
	github.com/cloudflare/terraform-provider-cloudflare/internal/provider/resource_cloudflare_custom_ssl.go:151 +0x57f
github.com/cloudflare/terraform-provider-cloudflare/internal/provider.resourceCloudflareCustomSslImport({0x1013fe0, 0xc000728a80}, 0xc0000fa300, {0xe93f00, 0xc000328480})
	github.com/cloudflare/terraform-provider-cloudflare/internal/provider/resource_cloudflare_custom_ssl.go:196 +0x35e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc0000b4300, {0x1013fe0, 0xc000728a80}, 0xc000913698, {0xc0005aa000, 0x45})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/provider.go:385 +0x1ac
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc000010078, {0x1013fe0?, 0xc000728960?}, 0xc0007300c0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/grpc_provider.go:1088 +0xf4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ImportResourceState(0xc000252a00, {0x1013fe0?, 0xc0007281e0?}, 0xc00072a0a0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:848 +0x1f6
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler({0xe63e00?, 0xc000252a00}, {0x1013fe0, 0xc0007281e0}, 0xc000732000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:403 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00049e000, {0x1017be0, 0xc0003c3040}, 0xc000122b40, 0xc00058ef00, 0x156ef18, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1295 +0xb2b
google.golang.org/grpc.(*Server).handleStream(0xc00049e000, {0x1017be0, 0xc0003c3040}, 0xc000122b40, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1636 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.48.0/server.go:930 +0x28a

Error: The terraform-provider-cloudflare_v3.25.0 plugin crashed!
```

--- a/.changelog/1964.txt
+++ b/.changelog/1964.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_custom_ssl: check GeoRestrictions is not nil before attempting to compare it
+```

--- a/internal/provider/resource_cloudflare_custom_ssl.go
+++ b/internal/provider/resource_cloudflare_custom_ssl.go
@@ -275,7 +275,7 @@ func flattenCustomSSLOptions(sslopt cloudflare.ZoneCustomSSLOptions) map[string]
 		"type":          sslopt.Type,
 	}
 
-	if sslopt.GeoRestrictions.Label != "" && sslopt.GeoRestrictions.Label != "custom" {
+	if sslopt.GeoRestrictions != nil && sslopt.GeoRestrictions.Label != "" && sslopt.GeoRestrictions.Label != "custom" {
 		data["geo_restrictions"] = sslopt.GeoRestrictions.Label
 	}
 

